server/src/main/java/org/elasticsearch/rest/action/search/RestExplainAction.java

```diff
@@ -40,10 +42,20 @@
  * Rest action for computing a score explanation for specific documents.
  */
 public class RestExplainAction extends BaseRestHandler {
+    private static final DeprecationLogger deprecationLogger = new DeprecationLogger(
+        LogManager.getLogger(RestExplainAction.class));
+    static final String TYPES_DEPRECATION_MESSAGE = "[types removal] " +
+        "Specifying a type in explain requests is deprecated.";
+
     public RestExplainAction(Settings settings, RestController controller) {
         super(settings);
         controller.registerHandler(GET, "/{index}/{type}/{id}/_explain", this);
         controller.registerHandler(POST, "/{index}/{type}/{id}/_explain", this);
+
+        // This path actually represents "/{index}/{id}/_explain". Because of limitations in the path
```

Comment:
After consulting with the core-infra team, they confirmed the limitations our REST controller and suggested this approach.